lib/transporters.js

```diff
@@ -73,29 +73,28 @@ DefaultTransporter.prototype.wrapCallback_ = function(opt_callback) {
       body = JSON.parse(body);
     } catch (err) { /* no op */ }
 
-    if (body && body.error) {
-      // handle single request errors
+    // Consider the response an error if it includes the 'error' property or it has a
+    // 500 status code.
+    if ((body && body.error) || res.statusCode >= 500) {
```

Comment:
I don't really like this nested if block. I believe we could achieve the same thing with an if-else block